spec/furniture/marketplace/stripe_events_controller_request_spec.rb

```diff
@@ -81,6 +82,113 @@
         expect(Stripe::Transfer).not_to(have_received(:create))
       end
     end
+
+    context "when Square notifications are not enabled" do
+      let(:marketplace) { create(:marketplace) }
+      let(:person) { create(:person) }
+      let(:shopper) { create(:marketplace_shopper, person: person) }
+      let(:order) { create(:marketplace_order, :with_products, status: :pre_checkout, marketplace: marketplace, shopper: shopper) }
+
+      it "does not attempt to transfer the order to seller's Square dashboard" do
+        square_order_double = instance_double(Marketplace::SquareOrder, send_order: true)
```

Comment:
I would probably test the negative case against the `fake_square`; because right now there's not really any way for this test to fail; since the `square_order_double` is local to this block; and I'm not seeing a place where you're injecting this double into the underlying call stack.

For example:

```rb
fake_square = instance_double(Square::Client, :orders)`
allow(Square::Client).to receive(:new).and_return(:fake_square)
call
expect(fake_square).not_to have_received(:orders)
```

Reply:
I'm not sure I understand the difference, mainly because I just care about testing the conditional logic in `marketplace.square_connection` and this test achieves that goal. Are you interested in also covering failure mode(s) in instantiation of the `SquareOrder`?

Reply:
If you try and make your current test fail; It can't.  At no point does the fake get accessed by the method-under-test.

If you don't want to switch to testing against the `Square::Client` (the "inner-most-outer-edge" of the seam between Convene and Square); you would still need to do something like:


```rb
square_order_double = instance_double(Marketplace::SquareOrder, send_order: true)
# Inject our double into the call stack
allow(Marketplace::SquareOrder).to receive(:new).and_return(square_order_double) 
call
expect(square_order_double).not_to have_received(:send_order)
```



Both are perfectly acceptable;

Reply:
Ehh, I'm still getting things confused because of JavaScript and forgetting about needing to invoke `new` and _allowing_ that into the call stack (so to speak). 

I'm not as concerned with testing down to the `Square::Client` with this negative test since we don't even get that far.